File: src/main/java/easy/RelativeSortArray.java
package easy;

import java.util.HashMap;
import java.util.List;

public class RelativeSortArray {
    public int[] solve(int[] arr1, int[] arr2) {
        int[] result = new int[arr1.length];
        int index = 0;
        HashMap<Integer, Integer> freq = new HashMap<>();
        for (int i : arr1) {
            freq.put(i, freq.getOrDefault(i, 0) + 1);
        }
        for (int i : arr2) {
            int count = freq.getOrDefault(i, 0);
            while (count > 0) {
                result[index++] = i;
                count--;
            }
            freq.remove(i);
        }
        List<Integer> sorted = freq.keySet().stream().sorted().toList();
        for (int i : sorted) {
            int count = freq.getOrDefault(i, 0);
            while (count > 0) {
                count--;
                result[index++] = i;
            }
        }
        return result;
    }
}

File: src/easy/RelativeSortArray.swift
func relativeSortArray(_ arr1: [Int], _ arr2: [Int]) -> [Int] {
  var result = Array(repeating: 0, count: arr1.count)
  var freq: [Int: Int] = [:]
  var index = 0
  for num in arr1 {
    freq[num, default: 0] += 1
  }
  for num in arr2 {
    var count = freq[num, default: 0]
    while count > 0 {
      result[index] = num
      count -= 1
      index += 1
    }
    freq[num] = nil
  }

  for key in freq.keys.sorted() {
    if var current = freq[key] {
      while current > 0 {
        result[index] = key
        index += 1
        current -= 1
      }

    }
  }
  return result
}
